server/api/api_test.go

```diff
@@ -168,3 +170,120 @@ func TestSetLink(t *testing.T) {
 		})
 	}
 }
+
+func TestGetLink(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		prevLinks    []autolink.Autolink
+		autoLinkName string
+		expectStatus int
+		expectReturn string
+	}{
+		{
+			name:         "get the autolink",
+			autoLinkName: "test",
+			prevLinks: []autolink.Autolink{{
+				Name:     "test",
+				Pattern:  ".*1",
+				Template: "test",
+			}},
+			expectStatus: http.StatusOK,
+			expectReturn: `{"Name":"test","Disabled":false,"Pattern":".*1","Template":"test","Scope":null,"WordMatch":false,"DisableNonWordPrefix":false,"DisableNonWordSuffix":false,"ProcessBotPosts":false}`,
+		},
+		{
+			name:         "not found",
+			autoLinkName: "test",
+			prevLinks: []autolink.Autolink{{
+				Name:     "test1",
+				Pattern:  ".*1",
+				Template: "test",
+			}},
+			expectStatus: http.StatusInternalServerError,
+			expectReturn: "{\"error\":\"An internal error has occurred. Check app server logs for details.\",\"details\":\"no autolink found with name test\"}",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			var saved []autolink.Autolink
+			var saveCalled bool
+
+			h := NewHandler(
+				&linkStore{
+					prev:       tc.prevLinks,
+					saveCalled: &saveCalled,
+					saved:      &saved,
+				},
+				authorizeAll{},
+			)
+
+			w := httptest.NewRecorder()
+			r, err := http.NewRequest(http.MethodGet, fmt.Sprintf("/api/v1/link?autolinkName=%s", tc.autoLinkName), nil)
+			require.NoError(t, err)
+
+			r.Header.Set("Mattermost-Plugin-ID", "testfrom")
+			r.Header.Set("Mattermost-User-ID", "testuser")
```

Comment:
same

Reply:
check everywhere

Reply:
They have not used any constant anywhere